libs/components/core/testing/src/media-query/media-query-testing.service.ts

```diff
@@ -0,0 +1,142 @@
+  SkyMediaQueryListener,
+  SkyMediaQueryServiceInterface,
+  SkyResizeObserverMediaQueryService,
+  SkyResizeObserverMediaQueryServiceInterface,
+  toSkyMediaBreakpoints,
+} from '@skyux/core';
+
+import {
+  BehaviorSubject,
+  Observable,
+  Subscription,
+  firstValueFrom,
+} from 'rxjs';
+
+import { SkyMediaQueryTestingController } from './media-query-testing.controller';
+
+/**
+ * @internal
+ */
+@Injectable({ providedIn: 'root' })
+export class SkyMediaQueryTestingService
+  implements
+    SkyMediaQueryServiceInterface,
+    SkyResizeObserverMediaQueryServiceInterface,
+    SkyMediaQueryTestingController,
+    OnDestroy
+{
+  public get breakpointChange(): Observable<SkyMediaBreakpointType> {
+    return this.#breakpointChangeObs;
+  }
+
+  public get current(): SkyMediaBreakpoints {
+    return this.#currentBreakpoint;
+  }
+
+  #breakpointChange = new BehaviorSubject<SkyMediaBreakpointType>(
+    SKY_MEDIA_BREAKPOINT_TYPE_DEFAULT,
+  );
+  #breakpointChangeObs = this.#breakpointChange.asObservable();
+
+  #currentBreakpoint = SKY_MEDIA_BREAKPOINT_DEFAULT;
+  #currentSubject = new BehaviorSubject<SkyMediaBreakpoints>(
+    SKY_MEDIA_BREAKPOINT_DEFAULT,
+  );
+
+  #observeSubscription: Subscription | undefined;
+
+  readonly #renderer = inject(RendererFactory2).createRenderer(undefined, null);
+
+  public ngOnDestroy(): void {
+    this.destroy();
+  }
+
+  public destroy(): void {
+    this.#observeSubscription?.unsubscribe();
+    this.#currentSubject.complete();
+    this.#breakpointChange.complete();
+  }
+
+  public async expectBreakpoint(
+    expectedBreakpoint: SkyMediaBreakpointType,
+  ): Promise<void> {
+    const current = await firstValueFrom(this.#breakpointChange);
+
+    if (expectedBreakpoint !== current) {
+      throw new Error(
+        `Expected the current media breakpoint to be "${expectedBreakpoint}", but it is "${current}".`,
+      );
+    }
+  }
+
+  public setBreakpoint(breakpoint: SkyMediaBreakpointType): void {
+    const breakpointLegacy = toSkyMediaBreakpoints(breakpoint);
+    this.#currentBreakpoint = breakpointLegacy;
+    this.#currentSubject.next(breakpointLegacy);
+    this.#breakpointChange.next(breakpoint);
+  }
+
+  public subscribe(listener: SkyMediaQueryListener): Subscription {
+    return this.#currentSubject.subscribe({
+      next: (breakpoint) => {
+        listener(breakpoint);
+      },
+    });
+  }
+
+  public observe(
+    elementRef: ElementRef,
+    options?: { updateResponsiveClasses?: boolean },
+  ): SkyResizeObserverMediaQueryService {
+    this.#observeSubscription?.unsubscribe();
+    this.#observeSubscription = this.#breakpointChangeObs.subscribe(
+      (breakpoint) => {
+        if (options?.updateResponsiveClasses) {
+          const el = elementRef.nativeElement;
+
+          this.#removeResponsiveClasses(el);
+          this.#addResponsiveClass(el, breakpoint);
+        }
+      },
+    );
+
+    return this as unknown as SkyResizeObserverMediaQueryService;
```

Comment:
This makes me think we're overloading the purpose of this service. Why not have a separate `SkyResizeObserverMediaQueryTestingService`?

Reply:
We're overloading the `SkyMediaQueryService` injection token with an instance of `SkyResizeObserverMediaQueryService`. In other words, we're never _really_ injecting `SkyResizeObserverMediaQueryService` into our components, we're only overloading the `SkyMediaQueryService` token with the functionality we want (and asserting the type as `SkyResizeObserverMediaQueryService` so we can call its methods).

For unit tests, the `SkyMediaQueryService` token is overridden with the `SkyMediaQueryTestingService`, so it also needs to include the public methods from the resize observer service.

```
providers: [
  {
    provide: SkyMediaQueryService,
    useClass: SkyResizeObserverMediaQueryService
              ^
              This class has public methods that aren't defined in SkyMediaQueryService
  }
]
```

```
// Inject the `SkyMediaQueryService` token, but assert its type as `SkyResizeObserverMediaQueryService`
// so we can access its public methods.
#querySvc = inject(SkyMediaQueryService) as SkyResizeObserverMediaQueryService;
```